examples/swapAndBridge.ts

```diff
@@ -0,0 +1,182 @@
+import { Order, OrderBalance, OrderKind } from '@cowprotocol/contracts';
+import {
+  ABI_CODER,
+  USDC,
+  VAULT_RELAYER,
+  WETH,
+  approveToken,
+  createOrder,
+  fnCalldata,
+  settleOrder,
+  withAnvilProvider,
+  wrapEther,
+} from './common';
+import { ethers } from 'ethers_v6';
+import { CowShedSdk, ICall } from '../ts';
+
+const GNOSIS_CHAIN_BRIDGE = '0x88ad09518695c6c3712AC10a214bE5109a655671';
+
+const swapAndBridge: Parameters<typeof withAnvilProvider>[0] = async (
+  provider: ethers.JsonRpcProvider,
+  signers: ethers.Wallet[],
+  { factory, implementation, initCode }
+) => {
+  const user = signers[1];
+  const userAddr = await user.getAddress();
+
+  // wrap the ether and approve to cowswap
+  const amt = ethers.parseEther('1');
+  const wrapTx = await wrapEther(provider, userAddr, amt);
+  console.log('ether wrapped', wrapTx?.hash);
+  const approveTx = await approveToken(
+    provider,
+    WETH,
+    userAddr,
+    VAULT_RELAYER,
+    amt
+  );
+  console.log('weth approved', approveTx?.hash);
+
+  // compute the proxy address with CowShedSdk
+  const shedSdk = new CowShedSdk({
+    factoryAddress: factory,
+    implementationAddress: implementation,
+    chainId: 1,
+    proxyCreationCode: initCode,
+  });
+  const proxyAddress = shedSdk.computeProxyAddress(userAddr);
+  console.log('Proxy address for user', userAddr, 'is', proxyAddress);
+
+  const buyAmount = 10_000000n;
+  const validTo = Math.floor(new Date().getTime() / 1000) + 7200;
+
+  const order: Order = {
+    sellToken: WETH,
+    buyToken: USDC,
+    receiver: proxyAddress,
+    sellAmount: amt,
+    buyAmount,
+    validTo,
+    appData: '',
+    feeAmount: 0n,
+    kind: OrderKind.SELL,
+    partiallyFillable: true,
+    sellTokenBalance: OrderBalance.ERC20,
+    buyTokenBalance: OrderBalance.ERC20,
+  };
+
+  // post hooks
+  const calls: ICall[] = [
+    // approve the bridge to spend the swapped usdc
+    {
+      target: USDC,
+      callData: fnCalldata(
+        'approve(address,uint256)',
+        ABI_CODER.encode(['address', 'uint'], [GNOSIS_CHAIN_BRIDGE, buyAmount])
```

Comment:
nit: note that as it is a `sell` order, there may be at _least_ `buyAmount`, but there may actually be _more_ than `buyAmount`...

Reply:
Additionally, please make consistent use of specifying the bitwidth of `uint` (i.e. no `uint`, use `uint256` :pray: )

Reply:
yeah, was a conscious choice, surplus will just stay in the user controlled proxy